src/plugin/runtime-support.cc

```diff
@@ -427,49 +429,98 @@ void rt::protocol_begin(const std::optional<std::string>& analyzer) {
 
         if ( ! c->analyzer->AddChildAnalyzer(child) )
             // Same comment as above re/ ignoring the error and memory mgmt.
-            return;
+            return child->GetID();
 
         child->FirstPacket(true, nullptr);
         child->FirstPacket(false, nullptr);
+
+        return {};
     }
 }
 
-void rt::protocol_data_in(const hilti::rt::Bool& is_orig, const hilti::rt::Bytes& data) {
+void rt::protocol_data_in(const hilti::rt::Bool& is_orig, const hilti::rt::Bytes& data,
+                          const std::optional<rt::ProtocolHandle>& h) {
     auto cookie = static_cast<Cookie*>(hilti::rt::context::cookie());
     assert(cookie);
 
     auto c = std::get_if<cookie::ProtocolAnalyzer>(cookie);
     if ( ! c )
         throw ValueUnavailable("no current connection available");
 
-    c->analyzer->ForwardStream(data.size(), reinterpret_cast<const u_char*>(data.data()), is_orig);
+    auto len = data.size();
+    auto* data_ = reinterpret_cast<const u_char*>(data.data());
+
+    if ( h ) {
+        if ( auto* output_handler = c->analyzer->GetOutputHandler() )
+            output_handler->DeliverStream(len, data_, is_orig);
+
+        // Forward empty payload to trigger lifecycle management in this analyzer tree.
+        c->analyzer->ForwardStream(0, data_, is_orig);
+
+        auto* child = c->analyzer->FindChild(h->id());
+        if ( ! child || child->IsFinished() || child->Removing() )
+            throw ValueUnavailable(hilti::rt::fmt("unknown child analyzer %s", *h));
```

Comment:
Split into two error cases: "unknown analyzer" and "no longer exists"; might help tracking down issues.

(Same below).